### ISSUE TITLE
Lookahead has no attribute 'state'

### ISSUE THREAD

**Issue by staylor-ds** [mgrankin/over9000#6]
When trying to save a model, such as in the mt-dnn module I encountered the following traceback.

```
    File "~mt-dnn/mt_dnn/model.py", line 263, in save
        optimizer.state_dict(),
    File "~/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/optim/optimizer.py", line 90, in state_dict
        for k, v in self.state.items()}
    AttributeError: 'Lookahead' object has no attribute 'state'
```
It appears that the Lookahead class is missing a call to super in it's `__init__` method (as seen in the other Optimizers):
```super(Lookahead, self).__init__(params, defaults)```
Although I am not sure if this will save the state of the wrapped optimizer?

**Comment by mgrankin** [comment]
`super(Lookahead, self).__init__(params, defaults)` - This shouldn't be in Lookahead class because it will mess with param_groups. param_groups are managed by base_optimizer.

I've tried to add following code. It will allow to save optimizer and base_optimizer, but that is definitely is not enough, I can't figure out what is missing right now.

`    def __getstate__(self):
        return self.__dict__
`

**Comment by mgrankin** [comment]
fixed here 613cf716a5ec205c79f4d130f0569b1c167571a0 by @rwightman
